File: app/src/main/java/me/tagavari/airmessage/util/ConversationTarget.java
package me.tagavari.airmessage.util;

import java.util.List;

/**
 * Represents a conversation that can be sent to
 */
public abstract class ConversationTarget {
	public static class AppleLinked extends ConversationTarget {
		private final String guid;
		
		public AppleLinked(String guid) {
			this.guid = guid;
		}
		
		public String getGuid() {
			return guid;
		}
	}
	
	public static class AppleUnlinked extends ConversationTarget {
		private final List<String> members;
		private final String service;
		
		public AppleUnlinked(List<String>members, String service) {
			this.members = members;
			this.service = service;
		}
		
		public List<String> getMembers() {
			return members;
		}
		
		public String getService() {
			return service;
		}
	}
	
	public static class SystemSMS extends ConversationTarget {
		private final long externalID;
		
		public SystemSMS(long externalID) {
			this.externalID = externalID;
		}
		
		public long getExternalID() {
			return externalID;
		}
	}
}

File: common/me/tagavari/airmessage/common/util/ConversationTarget.kt
package me.tagavari.airmessage.common.util

/**
 * Represents a conversation that can be sent to
 */
interface ConversationTarget {
	//A serverlinked Apple Messages chat
	data class AppleLinked(val guid: String) : ConversationTarget
	
	//An unlinked potential Apple Messages chat
	data class AppleUnlinked(val members: List<String>, val service: String) : ConversationTarget
	
	//A local SMS chat
	data class SystemSMS(val externalID: Long) : ConversationTarget
}